starlette/routing.py

```diff
@@ -591,27 +566,13 @@ def __init__(
         self.on_shutdown = [] if on_shutdown is None else list(on_shutdown)
 
         if lifespan is None:
-            self.lifespan_context: typing.Callable[
-                [typing.Any], typing.AsyncContextManager
-            ] = _DefaultLifespan(self)
-
-        elif inspect.isasyncgenfunction(lifespan):
-            warnings.warn(
-                "async generator function lifespans are deprecated, "
-                "use an @contextlib.asynccontextmanager function instead",
-                DeprecationWarning,
-            )
-            self.lifespan_context = asynccontextmanager(
-                lifespan,  # type: ignore[arg-type]
-            )
-        elif inspect.isgeneratorfunction(lifespan):
-            warnings.warn(
-                "generator function lifespans are deprecated, "
-                "use an @contextlib.asynccontextmanager function instead",
-                DeprecationWarning,
-            )
-            self.lifespan_context = _wrap_gen_lifespan_context(
-                lifespan,  # type: ignore[arg-type]
+            self.lifespan_context: Lifespan = _DefaultLifespan(self)
+        elif inspect.isasyncgenfunction(lifespan) or inspect.isgeneratorfunction(
+            lifespan
+        ):
+            raise RuntimeError(
+                "Generator functions are not supported for lifespan, "
+                "use an @contextlib.asynccontextmanager function instead."
```

Comment:
If we'ere removing deprecated things, why not just type this as a contextmanager and just hope it quacks like a duck?

Reply:
You mean just removing the conditional?

Reply:
I'm going to readd them